src/System.Binary.Base64/System/Binary/BufferExtensions.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers;
+using System.Collections.Sequences;
+
+// This should most likely be in System.Text.Primitives and not here. Will move once Base64 moves.
+namespace System.Binary.Base64
+{
+    public static class BufferExtensions
+    {
+        const int stackLength = 32;
+
+        public static void Pipe(this ITransformation transformation, ReadOnlyBytes source, IOutput destination)
```

Comment:
I think IOutput and ROB are fine for now. But we need to close on whether these are the primitives we want.
ROB needs to be unified with ReadableBuffer
IOutput: we need to validate that it works for chaining transformers.